src/renderers/dom/client/wrappers/ReactDOMInput.js

```diff
@@ -140,13 +138,14 @@ var ReactDOMInput = {
 
     var value = LinkedValueUtils.getValue(props);
     if (value != null) {
-      // Cast `value` to a string to ensure the value is set correctly. While
-      // browsers typically do this as necessary, jsdom doesn't.
-      DOMPropertyOperations.setValueForProperty(
-        ReactDOMComponentTree.getNodeFromInstance(inst),
-        'value',
-        '' + value
-      );
+      var node = ReactDOMComponentTree.getNodeFromInstance(inst)
+
+      // To avoid side effects (such as losing text selection), only set value if changed
+      if (value != node.value) {
```

Comment:
Previously, the string cast happened before this comparison, so `value` of `3` and `node.value` of `'3'` would not trigger a set. We should keep that.